homeassistant/components/timer.py

```diff
@@ -0,0 +1,261 @@
+        target_timers = component.async_extract_from_service(service)
+
+        attr = None
+        if service.service == SERVICE_START:
+            attr = 'async_start'
+        elif service.service == SERVICE_PAUSE:
+            attr = 'async_pause'
+
+        tasks = [getattr(timer, attr)() for timer in target_timers if attr]
+        if service.service == SERVICE_RESET:
+            for timer in target_timers:
+                tasks.append(
+                    timer.async_reset(service.data.get(ATTR_DURATION,
+                                                       DEFAULT_INITIAL)))
+        if tasks:
+            yield from asyncio.wait(tasks, loop=hass.loop)
+
+    descriptions = yield from hass.async_add_job(
+        load_yaml_config_file, os.path.join(
+            os.path.dirname(__file__), 'services.yaml')
+    )
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_START, async_handler_service,
+        descriptions[DOMAIN][SERVICE_START], SERVICE_SCHEMA)
+    hass.services.async_register(
+        DOMAIN, SERVICE_PAUSE, async_handler_service,
+        descriptions[DOMAIN][SERVICE_PAUSE], SERVICE_SCHEMA)
+    hass.services.async_register(
+        DOMAIN, SERVICE_RESET, async_handler_service,
+        descriptions[DOMAIN][SERVICE_RESET], SERVICE_SCHEMA_RESET)
+
+    yield from component.async_add_entities(entities)
+    return True
+
+
+class Timer(Entity):
+    """Representation of a timer."""
+
+    def __init__(self, hass, object_id, name, initial, icon):
+        """Initialize a timer."""
+        self.entity_id = ENTITY_ID_FORMAT.format(object_id)
+        self._name = name
+        self._state = self._initial = initial
+        self._icon = icon
+        self._status = STATUS_IDLE
+        self._hass = hass
+
+        async_track_time_interval(
+            hass, self.async_update, INTERVAL
+        )
+
+    @property
+    def should_poll(self):
+        """If entity should be polled."""
+        return False
+
+    @property
+    def name(self):
+        """Return name of the timer."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to be used for this entity."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the current value of the timer."""
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_INITIAL: self._initial,
+            ATTR_STATUS: self._status,
+        }
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Call when entity is about to be added to Home Assistant."""
+        # If not None, we got an initial value.
+        if self._state is not None:
+            return
+
+        state = yield from async_get_last_state(self.hass, self.entity_id)
+        self._state = state and state.state == state
+
+    @asyncio.coroutine
+    def async_start(self):
```

Comment:
need not to be a coro, you can use `self.async_schedule_update_ha_state()` that are a normal function